[SandboxIR] Add more functions to sandboxir:Instruction class.

tmsri opened the pull request:
The getter functions simply turn around and call the LLVM
counterparts.  This is fine until we don't add new sandbox IR
opcodes.  At that point, we may have to explicitly check if
the behavior is different.

tmsri committed:
Add more functions to sandboxir:Instruction class.

The getter functions simply turn around and call the LLVM
counterparts.  This is fine until we don't add new sandbox IR
opcodes.  At that point, we may have to explicitly check if
the behavior is different.

tmsri committed:
clang formatting.

Sterling-Augustine reviewed (approved):
Pretty straight-forward pass through functions.

vporpo commented on llvm/include/llvm/SandboxIR/SandboxIR.h:
```diff
@@ -1954,6 +1962,55 @@ class Instruction : public sandboxir::User {
+  //===--------------------------------------------------------------------===//
+
+  /// Return true if the instruction has any metadata attached to it.
+  bool hasMetadata() const { return cast<llvm::Instruction>(Val)->hasMetadata(); }
```
Not sure about whether we should give access to metadata at this point, because these are LLVM IR structures and we can't really track their state.

tmsri replied:
Removed getMetadata and getAllMetadata functions.

tmsri committed:
Remove getMetadata and getAllMetadata.

Add them back after MDNode is available in sandboxir.  For now,
exposing them could give access to functions that modify the
metadata.

vporpo reviewed (approved):
Looks good, thanks!

vporpo commented:
This should fix it: 13809b3d95924d691db53e62182d8ad46b0e08ec
